dashboard/app/models/user.rb

```diff
@@ -1159,28 +1159,48 @@ def in_progress_and_completed_scripts
     end
   end
 
-  # Return a collection of courses and scripts for the user. First in the list will
-  # be courses enrolled in by the user's sections. Following that will be all scripts
-  # in which the user has made progress that are not in any of the enrolled courses.
-  def recent_courses_and_scripts(exclude_primary_script)
-    courses = section_courses
-    course_scripts_script_ids = courses.map(&:default_course_scripts).flatten.pluck(:script_id).uniq
-
-    # filter out those that are already covered by a course
-    user_scripts = in_progress_and_completed_scripts.
-      select {|user_script| !course_scripts_script_ids.include?(user_script.script_id)}
-
-    course_data = courses.map do |course|
+  # Returns an array of hashes storing data for each unique course assigned to # sections that this user is a part of.
+  # @returns [Array{CourseData}]
+  def assigned_courses
+    assigned_courses_data = section_courses.map do |course|
       {
         name: course[:name],
         title: data_t_suffix('course.name', course[:name], 'title'),
         description: data_t_suffix('course.name', course[:name], 'description_short'),
         link: course_path(course),
       }
     end
+    assigned_courses_data
+  end
+
+  # Checks if there are any non-hidden scripts assigned to the user.
+  # @returns boolean
+  def any_visible_assigned_scripts?
+    assigned_user_scripts = user_scripts.where("assigned_at")
 
+    assigned_visible_scripts = assigned_user_scripts.map {|user_script| Script.where(id: user_script.script_id, hidden: 'false')}
+
+    !assigned_visible_scripts.flatten.empty?
+  end
+
+  # Checks if there are any non-hidden scripts or courses assigned to the user.
+  # @returns boolean
+  def assigned_course_or_script?
+    !assigned_courses.empty? || any_visible_assigned_scripts?
```

Comment:
I think you could also use `assigned_courses.any?` here instead of `!assigned_courses.empty?`. Not sure if people with more ruby knowledge like @aoby have guidance here - I think one place .any? can get you in trouble is that `[false].any?` is false (i.e. it looks for any truthy value).

Reply:
Agreed, `any?` is preferable to `!empty?`, and presumably assigned_courses will all be models, and therefore truthy.